modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/services/config/ConfigurationServiceImpl.java

```diff
@@ -59,32 +59,8 @@
 
     protected static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
-    /**
-     * XXX remove once we are able to get such a cached map from DefaultComponent
-     *
-     * @since 10.3
-     */
-    protected volatile Map<String, ConfigurationPropertyDescriptor> descriptors;
-
-    /**
-     * XXX remove once we are able to get such a cached map from DefaultComponent.
-     * <p>
-     * We'd ideally need a <T extends Descriptor> Map<String, T> getDescriptors(String xp) with cache method.
-     *
-     * @since 10.3
-     */
-    protected Map<String, ConfigurationPropertyDescriptor> getDescriptors() {
-        Map<String, ConfigurationPropertyDescriptor> d = descriptors;
-        if (d == null) {
-            synchronized (this) {
-                d = descriptors;
-                if (d == null) {
-                    List<ConfigurationPropertyDescriptor> descs = getDescriptors(CONFIGURATION_EP);
-                    descriptors = d = descs.stream().collect(Collectors.toMap(desc -> desc.getId(), desc -> desc));
-                }
-            }
-        }
-        return d;
+    protected MapRegistry getMapRegistry() {
+        return getRegistry(CONFIGURATION_EP, MapRegistry.class);
     }
```

Comment:
I think we should have the same method on `MapRegistry` and `SingleRegistry`, depending on the implementation an exception may be thrown. Then this is the service's responsibility to know which methods it has to call, and this will avoid such methods.

Reply:
I started by putting both Map and Single API on the interface indeed, but i'm not sure it's clearer, and since people can define their own registry class, maybe it's better to keep this out of the interface (?). We can rediscuss this in all cases, i'll try to re-arrange the code with other feedback first.

Reply:
I think API should be simpler now, but we can discuss this again in the upcoming migration PR (will reference it here when done)